muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/rest/ForumRESTService.java

```diff
@@ -967,7 +984,7 @@ public Response toggleForumThreadSubscription(@PathParam ("AREAID") Long areaId,
         return Response.status(Status.NOT_FOUND).entity("Forum thread not found from the specified area").build();
       }
       
-      if (forumThread.getLocked()) {
+      if (forumThread.getLock().equals(LockForumThread.ALL) || (forumThread.getLock().equals(LockForumThread.STUDENTS) && userEntityController.isStudent(sessionController.getLoggedUserEntity()))) {
```

Comment:
Use `==` to avoid `NullPointerException` if `forumThread().getLock()` would happen to be `null`.